tiledb/sm/storage_manager/context.cc

```diff
@@ -210,35 +214,33 @@ size_t Context::get_io_thread_count(const Config& config) {
       std::max(config_thread_count, io_concurrency_level));
 }
 
-Status Context::init_loggers(const Config& config) {
+void Context::init_loggers(const Config& config) {
   // temporarily set level to error so that possible errors reading
   // configuration are visible to the user
   logger_->set_level(Logger::Level::ERR);
 
   const char* format_conf;
-  RETURN_NOT_OK(config.get("config.logging_format", &format_conf));
+  throw_if_not_ok(config.get("config.logging_format", &format_conf));
   assert(format_conf != nullptr);
   Logger::Format format = Logger::Format::DEFAULT;
-  RETURN_NOT_OK(logger_format_from_string(format_conf, &format));
+  throw_if_not_ok(logger_format_from_string(format_conf, &format));
 
   global_logger(format);
   logger_->set_format(static_cast<Logger::Format>(format));
 
   // set logging level from config
   bool found = false;
   uint32_t level = static_cast<unsigned int>(Logger::Level::ERR);
-  RETURN_NOT_OK(config.get<uint32_t>("config.logging_level", &level, &found));
+  throw_if_not_ok(config.get<uint32_t>("config.logging_level", &level, &found));
   assert(found);
   if (level > static_cast<unsigned int>(Logger::Level::TRACE)) {
-    return logger_->status(Status_StorageManagerError(
+    throw std::invalid_argument(
```

Comment:
This is the wrong exception. `invalid_argument` is for bad arguments to functions; it's akin to `logic_error`. This is not the case here. We don't have `ContextException` defined yet. It's time to do so.